### ISSUE TITLE
Scheduling strategy enhancements about SpreadConstraints

### ISSUE THREAD

**Issue by whitewindmills** [karmada-io/karmada#3243]
**What would you like to be added**:
Add field `whenUnsatisfiable` indicates how to deal with a workload if it doesn't satisfy the spread constraints:
* `DoNotSchedule` : tells the karmada scheduler not to schedule it(it's default strategy).
* `ScheduleAnyway`: tells the karmada scheduler to schedule it anyway, but satisfy the spread constraints as much as possible.

**Why is this needed**:
The spread constraint is just an ideal result we set, and the specific environment may not be satisfactory, but from a user's point of view, we don't want the scheduling to terminate because of this. In addition, even if the spread constraint is satisfied during first scheduling, it is not necessarily satisfied during failover. Should the failover be terminated at this time?

**Comment by RainbowMango** [comment]
Currently, the spread constraint is kind of a `hard` restriction that must be satisfied during scheduling. 

> ScheduleAnyway: tells the karmada scheduler to schedule it anyway, but satisfy the spread constraints as much as possible.

For now, can we do that by more flexible spread constraints, like a larger [min, max] range?

**Comment by whitewindmills** [comment]
@RainbowMango 
Thanks for your feedback. I don't think it's a panacea.
For example, My nginx server needs to be deployed in at least three regions to ensure high availability, so I can set `minGroups` to 3. High availability cannot be satisfied temporarily if some regions fail, but my nginx server needs to be scheduled to available regions to provide the service regardless of whether `minGroups` is met.
In general, we need `ScheduleAnyway` to satisfy the soft restriction.

**Comment by RainbowMango** [comment]
I get it. You just want Karmada `try` to spread the workload to `3` regions. 
As far as know, If people want to spread their workload, there is always a strong demand that they must `ensure` that, otherwise, they must stop and fix the block issue.

I want to learn more about your use case.

**Comment by whitewindmills** [comment]
First, I want to make sure if you feel 'ScheduleAnyway' is necessary.

Of course, it is no problem to confirm how many regions there are before first scheduling. If rescheduling is triggered, such as `failover`, maybe the previous spread constraints are no longer applicable, but we hope that Karmada can continue to schedule at this time. Usually we don't want manual intervention to modify the spread constraints.

**Comment by RainbowMango** [comment]
> First, I want to make sure if you feel ScheduleAnyway is necessary.

I can't tell yet. Now we need to figure out if this is a common use case, then we'll start to work out the solution. Does this process make sense to you?

**Comment by whitewindmills** [comment]
> I can't tell yet. Now we need to figure out if this is a common use case, then we'll start to work out the solution. Does this process make sense to you?

Of course, it does make sense to me. I'll look forward to your further feedback.

**Comment by RainbowMango** [comment]
Yeah, I'm still thinking about this case.
